### ISSUE TITLE
backend.xml pulls the all scripts again, causing non deterministic JS errors

### ISSUE THREAD

**Issue by sneridagh** [plone/plonetheme.barceloneta#158]
This rule:
`    <after css:theme-children="head" css:content="head script" />`
is present in both rules.xml and in backend.xml, so that causes to be added twice:

<img width="812" alt="screenshot 2018-09-28 at 10 28 52" src="https://user-images.githubusercontent.com/486927/46197216-6fa4e680-c309-11e8-8872-b426f8c7e3b3.png">

In case that you are including backend.xml in your rule set.

**Comment by sneridagh** [comment]
I'll opened https://github.com/plone/bobtemplates.plone/issues/320 in bobtemplates.plone too.

**Comment by djay** [comment]
@sneridagh backend.xml is to switch between the a themed and "backend" theme versions of the site so problem is how backend.xml is included. It should replace any script and css includes in index.html, not be used in conjunction with it.

**Comment by sneridagh** [comment]
@djay I always used it the way that it's shown in the docs, and honestly, I think it's the more sensible way to do it (in terms of readability and approachability to newcomers).

Which is your approach, then? branch rules.xml to support both backend calls and non-backend ones? Then why to filter it again in backend.xml? I mean, you already have in backend.xml:

```
  <!-- Cut down barceloneta without just for backend UI -->
  <rules css:if-not-content="body.viewpermission-view, body.viewpermission-none">
```

**Comment by sneridagh** [comment]
@djay better thinking about it, I don't even think that we should include a backend.xml here in the p.barceloneta source code, since it's not used by it *at all*

I would vote for removing it, leave the one in bobtemplates.plone and improve the docs.

/cc @agitator @maikroeder @cdw9

**Comment by djay** [comment]
Sorry you are right. It was filtered inside backend.xml. 

So if it's an exclusive-or between backend or themed why would the js get included twice?  As long as your theme contains 

```
    <!-- Only theme front end pages -->
    <rules css:if-content="body.frontend#visual-portal-wrapper">
```
as per the docs.

Getting rid of it would be a bit of nasty backwards in compatibility. It would break all of our themes and plenty of others peoples too I would guess. I never use bobtemplates. 
Backend.xml is part of the barceloneta because the point is for custom themes to be able to rely on barceloneta to be there to theme the backend so they don't have to worry about it. That also means it should be tested. Ideally barceloneta should also come with another bundle that just includes whats needed for the backend only.

**Comment by sneridagh** [comment]
@djay good point, you can still pull it from there in your themes, so indeed removing is not an option.

I does get included twice (see the above screenshot), just both conditions are met, then both rules (the one in rules.xml and the one in backend.xml) are applied twice... I always thought that Diazo takes care of deduplication, but it seems not :( The issue go away as soon as you only include one.

**Comment by cdw9** [comment]
I've mostly avoided using backend.xml because it does not work consistently, and instead have opted to add styling for Site Setup. The issue I've run into with the latest version is that edit pages do not have the .frontend class, so backend.xml is used. This is likely the intended behavior, but I would prefer to have backend.xml only work within Site Setup.

So I guess my recommendation would be to remove it, and update docs to instruct how users can set it up themselves if they want it.

**Comment by djay** [comment]
@cdw9 so the bug is the make the body.frontend tag work consistently. The only inconstancy I had noticed was the history view because that is not protected by a permission. The @@edit page is certainly not considered backend so there is no inconsistency there. If you want it styled by your theme then you will need more complex rules.

Perhaps one confusing here is that .frontend is a not the best name. What is going on under the hood is that it is looking up what the permission the current view is protected by. You can be more explicit by using body.viewpermission-none, body.viewpermission-view instead of body.frontend. See https://github.com/plone/plone.app.layout/pull/50/files#diff-6f416e162a73a6671fbf3b692dd3ad7fR191. Then you could include .viewpermission-edit (or some other rule) to include @@edit pages in your themed pages.

Full discussion of this feature is in https://github.com/plone/Products.CMFPlone/issues/465

@sneridagh Looking at the code above they should be opposite from each other. given that .frontend == (.viewpermission-none or .viewpermission-view) so something weird must be going wrong with your rules. This is out of the box barceloneta?

**Comment by sneridagh** [comment]
@djay it is, in fact is the same that bundles in bobtemplates.plone themes, and as far as I remember it's like this since day one you made the PR back in the day.

**Comment by vincentfretin** [comment]
@mauritsvanrees From your comment https://github.com/plone/plonetheme.barceloneta/pull/159#issuecomment-618235656 can we close this issue?

**Comment by mauritsvanrees** [comment]
Yes, the fix is in `plonetheme.barceloneta 2.1.7`, included in 5.2.2. Closing.
